### PR TITLE
Fix 'close old issues' workflow

### PR DESCRIPTION
**Pull request by og-mrk**
### Changes for each commit

- ~Remove un-necessary permissions in 'Close Old Issues' GitHub Workflow. ( commit 8fe062421d32ca67d22a716f69260588ee118541 )~
  **Reverted in commit 6f838142a6674fcc9c4ad9acc402a4970840f496**
- Update 'Close Old Issues' GitHub Workflow to try and fix it, as there's currently a lot of inactive issues in the repo, and this commit adds quite a few comments which aids in future reviewing/understanding of this workflow file. ( commit 6e05426f125d6fc3f0ac94d749f7d583614c8281 )

### CODE REVIEW

File: .github/workflows/close-old-issues.yaml
Comment (og-mrk) [inline .github/workflows/close-old-issues.yaml:25]: I've written this because making a workflow have the ability to close any inactive PR (after a set period of time) is just.. it doesn't make any sense to me! At least in the context of WinUtil Repo + this workflow is designed for closing old issues (as file name suggests).

if anyone can find a compelling argument as to why we should expand the functionality of this workflow to close old PRs (with a period of time the same or different from issues).. then please go ahead and do so.

File: .github/workflows/close-old-issues.yaml
Comment (og-mrk) [inline .github/workflows/close-old-issues.yaml:24]: Had to split the time period of 2 weeks into:
- 1 week of inactivity -> mark issue as stale
- 1 week of inactivity (after marking the issue as stale) -> close issue

Reason behind this is because I don't know how to force this `actions/stale` GitHub Action to behave like the old JavaScript Code we had previously (close issue after 2 weeks of inactivity), if anyone knows a better way then this (while still using this GitHub Action).. then you go ahead and do so, test it, and make a PR for it 👍

**Comment by Marterich** [comment]
Maybe it would also be an option to setup a free [Github Organziation](https://docs.github.com/en/get-started/learning-about-github/githubs-plans#github-free-for-organizations), create Roles for the Repo and selectively allow frequent Maintainers to moderate the Issue and Discussion sections. If I'm correct, @ChrisTitusTech could still be the only one who is able to merge PRs but the rest of the Repo could be managed way better I think

**Comment by og-mrk** [comment]
> Maybe it would also be an option to setup a free [Github Organziation](https://docs.github.com/en/get-started/learning-about-github/githubs-plans#github-free-for-organizations), create Roles for the Repo and selectively allow frequent Maintainers to moderate the Issue and Discussion sections. If I'm correct, @ChrisTitusTech could still be the only one who is able to merge PRs but the rest of the Repo could be managed way better I think

Cool idea @Marterich, I think it's definitely worth the try, as managing issues, tracking requests (Feature and App requests) & bug reports, and reviewing PRs is.. it has some drag on my side if I'm being honest 😅

Now I don't really know how roles work in GitHub Organizations (haven't started one, or join any in the past), so anyone knowledgeable can _hopefully_ clarify this for me.. but if it only has a hand-full of Roles Presets with pre-defined permissions (can't create New Roles with customizable permissions that fits your needs).. then The best one I've found is the `Triage` Role ([link to the docs](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization), you'll find what each role has for permissions).. only problem(s) I have with the `Triage` Role is:

1. Can't create, edit (and even delete) Labels.. this will make managing issues a bit annoying, as you'll need to request to someone with the right permissions (Chris, or someone else) to create Labels and modify their name/description.. this will (at least for me) drag Issue Management.
   .. This one, after some thinking about, it isn't that bad.. _but I wish it was available_
2. You can close any Issue and **any PR**.. like what?!! **who would want that??**.. I can see this as a nightmare, if the person you gave this role (or any higher role) goes on there way to annoy the development process, giving WinUtil (as a Project and its community) a bad look.. I wouldn't allow anyone to close PRs no matter what (even IF the PR is a bad one).


image

**Comment by Marterich** [comment]
Fair points. To be totally honest, haven't set up one myself either and only glanced at the docs before so I didn't realize that only on enterprise you are able to create custom roles. 

Still "Organization owners can also choose to further limit access to certain settings and actions across the organization" is also mentioned so I guess we'll simply have to see what's possible. 

On the other hand, seeing that @ChrisTitusTech  would most likely would want to manually select maintainers anyway, there could be an agreement like "don't be stupid and merge any PRs or you lose the right immediately"

**Comment by og-mrk** [comment]
I know this GitHub Organizations thing should be its own separate subject to discuss (may be open a GitHub Issue and see other people suggestions and ideas).. but it still a small idea that will probably get rejected 😅 so I'll just share some links that might help us better understand it, and how to utilize it.

All the links point to GitHub Docs on different topics, but can be used together in the context of this idea/small proposal:
- GitHub Plans - GitHub Free for organizations: https://docs.github.com/en/get-started/learning-about-github/githubs-plans#github-free-for-organizations
- [Managing user access to your organization's repositories](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories) - _Highly Recommend at least browsing the other sections as well_
  - [Managing repository roles](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles)
    - [Repository roles for an organization](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization)
    - [About custom repository roles](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/about-custom-repository-roles)
- [About organization membership](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/about-organization-membership) - _A nice document to reference, definitely a good read._

</br>

@ChrisTitusTech @Marterich If you guys have any ideas or suggestions and want to share them, feel free to do so, and in the chance that it'll get Green-Lighted.. Me & @Marterich (if he want/have time) will come up with a proper proposal to the community, explaining the advantages of it in detail, what might change in the contribution process, and most importantly.. why do all it in the first place.

As a side note, although this might be like a little help in the managing of the repo, _that's in the short term_.. I believe it can better this project (and others that might arise from it) to a much higher scale then it's currently at, which's achievable with the help of contributors and maintainers worldwide.

Also Chris, if you like the idea and want to talk about it with the other Contributors (through GitHub Discussions, or other means like Email, Discord etc...), then I encourage you to do so, as even though you of course, as the project owner and maintainer, have the final say on it.. if no one want or can join in this effort.. then what's the point in all of it, other then add un-needed complexity :/

**Comment by og-mrk** [comment]
@ChrisTitusTech I've re-added the permissions in commit 6f838142a6674fcc9c4ad9acc402a4970840f496
I actually didn't know that Workflows had write permission to PRs by default 😅

**Review by ChrisTitusTech — APPROVED** [review]

**Comment by ChrisTitusTech** [comment]
looks great thanks @og-mrk

**Pull request merged**
